src/linalg/dense.jl

```diff
@@ -22,7 +22,24 @@ dense2arg = Dict{Symbol,Any}(
 # diag
 # diagm
 # trace
-# kron
+
+# code adapted from https://github.com/FluxML/Flux.jl/pull/169
+function _kron(mat1, mat2)
+    @assert ndims(mat1) == ndims(mat2) == 2 "Only support matrices for the time being"
+    m1, n1 = size(mat1)
+    mat1_rsh = reshape(mat1,(1,m1,1,n1))
+
+    m2, n2 = size(mat2)
+    mat2_rsh = reshape(mat2,(m2,1,n2,1))
+
+    return reshape(mat1_rsh.*mat2_rsh, (m1*m2,n1*n2))
+end
+
+kron(a::Rec, b::Rec)  = _kron(a, b)
+kron(a::Rec, b) = _kron(a, b)
+kron(a, b::Rec) = _kron(a, b)
+addtest(:kron, rand(2,3), rand(4,5))
```

Comment:
@CarloLucibello your implementation seems fully OK. But it may be better to have test case 
``` addtestN(:kron, rand(2,3), rand(4,5))``` instead of ``` addtest(:kron, rand(2,3), rand(4,5))```